### ISSUE TITLE
508-defect-2 [SCREENREADER]: Edit buttons MUST have an accessible label on the Review page

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#13014]
## [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description
Edit buttons in the ArrayField and ObjectField will have an `aria-label` set as `Edit undefined` when the `ui:title` and `schema.title` are undefined. It will also be set to `Edit  ` when the `ui:title` is set to a string with one space (a common practice).

Related: https://github.com/department-of-veterans-affairs/va.gov-team/issues/12955

## Acceptance Criteria
- [ ] As a screenreader user, the edit button should describe the content that it edits.

## Environment
* Operating System: All
* Browser: All
* Screenreading device: All
* Server: All

## Steps to Recreate

1. Log in to https://staging.va.gov/view-change-dependents/add-remove-form-686c/ as user 228
1. Check a box to report the death of a dependent, and one to remove a dependent
1. On the Review page, open all of the accordions
1. Inspect the "Edit" button aria-labels for "Edit undefined"

## Possible Fixes (optional)

- Add value from `ui:options.ariaLabelForEditButtonOnReview` (missing in `ArrayField`, but exists in `ObjectField`
- Fall back to `ui:title` >  `ui:options.reviewTitle` > schema title

## Screenshots or Trace Logs
![Screen Shot 2020-08-28 at 4 17 24 PM](https://user-images.githubusercontent.com/934879/91616175-d21a1d80-e94a-11ea-9605-843ad7937ffb.png)

## Definition of done

- [ ] Set a proper edit button aria-label (more than just "Edit", and not "Edit undefined")

**Comment by jason-gcio** [comment]
We are going to need some help with implementing the possible fixes shown in this ticket.  At first glance, it seems that this would be inside the Form System.

**Comment by jason-gcio** [comment]
After reviewing more of what Mottie did, we see [this](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/review/ObjectField.jsx#L152) from his [ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/12821).  We will see if we can do this.

**Comment by micahchiang** [comment]
Closing this ticket, since it appears that this issue has since been resolved. 

<details><summary>Edit buttons and their markups</summary>

<img width="1043" alt="Screen Shot 2020-12-03 at 11 27 49 AM" src="https://user-images.githubusercontent.com/15097156/101058519-373d6a80-355b-11eb-940f-5b6532db29cc.png">
<img width="980" alt="Screen Shot 2020-12-03 at 11 28 02 AM" src="https://user-images.githubusercontent.com/15097156/101058525-386e9780-355b-11eb-805a-2cecfb4b61d1.png">
<img width="1132" alt="Screen Shot 2020-12-03 at 11 28 46 AM" src="https://user-images.githubusercontent.com/15097156/101058531-399fc480-355b-11eb-8e01-756c6512b252.png">
<img width="1088" alt="Screen Shot 2020-12-03 at 11 28 58 AM" src="https://user-images.githubusercontent.com/15097156/101058533-3a385b00-355b-11eb-99b2-02d34c66e8cb.png">
<img width="966" alt="Screen Shot 2020-12-03 at 11 29 09 AM" src="https://user-images.githubusercontent.com/15097156/101058536-3ad0f180-355b-11eb-904b-2628541b3b14.png">

</details>

**Comment by 1Copenut** [comment]
Thanks @micahchiang this is looking good. We were having trouble with the buttons' accessible labels when there wasn't a heading directly beside them. Your last screenshot shows the expected outcome. 👍
